### ISSUE TITLE
[FindTeX] Parsing error with inline text nodes

### ISSUE THREAD

**Issue by ian-r-rose** [mathjax/mathjax-v3#78]
This is a bit subtle, and I am not entirely sure I have diagnosed the problem correctly, but I am running into a bug in the `findMath()` function in MathJax3. It appears that having nested elements in a paragraph (for instance, using the `<strong>` tag) causes premature exiting of the DOM string parser. Here is a minimal example, based upon the [`mj3-tex2html.html`](https://github.com/mathjax/mj3-demos/blob/master/mj3-tex2html.html) demo:
An HTML document with the following fails:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>Testing MathJax v3 setup with TeX input</title>
  <script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
  <script src="mj3-tex2html.dist.js"></script>
</head>
<body>
    <p>This is a <strong>spoiler</strong></p>
    <p>
    $$ E = mc^2 $$
    </p>
</body>
</html>
```
Producing
![image](https://user-images.githubusercontent.com/5728311/34457890-3c7dd412-ed74-11e7-8402-ddbb19d1b455.png)

Whereas the following succeeds: 
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>Testing MathJax v3 setup with TeX input</title>
  <script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
  <script src="mj3-tex2html.dist.js"></script>
</head>
<body>
    <p><strong>A spoiler</strong>, this ain't</p>
    <p>
    $$ E = mc^2 $$
    </p>
</body>
</html>
```
![image](https://user-images.githubusercontent.com/5728311/34457899-5d7c76c8-ed74-11e7-9b54-1303d730f306.png)

**Comment by pkra** [comment]
Thanks for the report.

The console throws an error `Uncaught TypeError: Cannot read property 'createElement' of null`.

**Comment by dpvc** [comment]
Thanks for catching this.  The problem was with keeping track of where to continue searching after looking at child nodes in a nested situation.  Having a nested node with no text following it was mishandled.  IT is an easy fix, which I have done in PR #79.

@pkra, the error message you point out is caused by there being no math found in this example; the document used for `document.createElement()` is not set in that case (because it gets set during the typesetting, but no typesetting occurred because there was no math found).  I have fixed that as well.  (This was also the source of the need for setting `chtml.node.document` in several of the example files, which should not have needed to be done, and this resolves that problem as well.)

**Comment by ian-r-rose** [comment]
Thanks for the quick fix! I have confirmed that #79 works for our use case (that is, markdown with inline TeX).

**Comment by dpvc** [comment]
Glad to hear that it resolves the issue for you.  Thanks again for the report.
